src/cli/java/org/commcare/util/cli/ApplicationHost.java

```diff
@@ -309,7 +309,7 @@ private boolean loopSession() throws IOException {
                         }
                     }
 
-                    screenIsRedrawing = !s.handleInputAndUpdateSession(mSession, input, false);
+                    screenIsRedrawing = !s.handleInputAndUpdateSession(mSession, input, false, null);

```

Comment:
@stephherbers this is the place where I anticipate changes for the multi-select referrals (You will need to work on top off this branch), I think we need to check here if we are on the `MultiSelectEntityScreen`. If yes, do following - 
 - Devise a strategy on how to take multiple cases as input from user from command prompts (probably ask user to enter all case ids at once in a tokenised string). Form a `selectedValues[]` with the user input
 - Check if we are on a `MultiSelectEntityScreen` and passing the input as `selectedValues[]` to the `handleInputAndUpdateSession` call.
 - print the `guid` corresponding to the saved `selectedValues[]` back on command prompt for user reference. 
 
 I have never ran CLI before myself so I might be slightly off in specifics of how it works but hopefully should not be far off from the general idea. Let me know if you have any questions and I would be happy to collab as well.